### PR TITLE
fix: add masking to log values

### PR DESCRIPTION
**Pull request by jonasflorencio**
## Proposed changes (including videos or screenshots)

The purpose of this PR is to mask sensitive settings data that was logged and exposed through the security-logs functionality, such as the `secret` and `password` types.

### Before:

image

### After:

image

## Issue(s)

[VLN-155
](https://rocketchat.atlassian.net/browse/VLN-155)

## Summary by CodeRabbit

* **New Features**
  * Audit logs now mask sensitive settings (passwords, secret-like values, usernames), showing limited characters for previous/current values while preserving actor/context across user/system/app actions.

* **Tests**
  * Added unit and end-to-end tests validating masking across updates, resets, actor types, and edge cases.

* **Chores**
  * Test configuration updated to include the new server test suites.

* **Public API**
  * Added a helper to read setting values for testing/restoration purposes.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/settings.ts [hunk 1]
```diff
@@ -423,5 +423,124 @@ describe('[Settings]', () => {
...
+					await updateSetting('SMTP_Password', originalSmtpPassword);
+				} else {
+					await updateSetting('SMTP_Password', '');
+				}
+
+				if (originalSmtpUsername !== undefined) {
+					await updateSetting('SMTP_Username', originalSmtpUsername);
+				} else {
+					await updateSetting('SMTP_Username', '');
+				}
```
Comment (KevLehman) [inline apps/meteor/tests/end-to-end/api/settings.ts]:
```suggestion
await updateSetting('SMTP_Password', originalSmtpPassword || '');
await updateSetting('SMTP_Username', originalSmtpUsername || '');
```
(linter will complain if u commit this cause of the tabs)

File: apps/meteor/server/settings/lib/auditedSettingUpdates.ts [hunk 2]
```diff
@@ -1,8 +1,44 @@
...
+		return value;
+	}
+
+	const valueString = String(value);
+	const valueLength = valueString.length;
+
+	let maskedValue: string;
+
+	if (valueLength <= 8) {
+		maskedValue = '*'.repeat(valueLength);
```
Comment (KevLehman) [inline apps/meteor/server/settings/lib/auditedSettingUpdates.ts:32]: Do we need the * to be the exact length? Can't we just put "*****" instead of repeat? 👀

File: apps/meteor/server/settings/lib/auditedSettingUpdates.ts [hunk 3]
```diff
@@ -1,8 +1,44 @@
...
+	} else {
+		const visiblePart = valueString.substring(0, 8);
+		const maskedPart = '*'.repeat(valueLength - 8);
+		maskedValue = visiblePart + maskedPart;
```
Comment (KevLehman) [inline apps/meteor/server/settings/lib/auditedSettingUpdates.ts]:
```suggestion
		maskedValue = `${valueString.substring(0, 8)}${'*****'}`
```

File: apps/meteor/server/settings/lib/auditedSettingUpdates.ts (hunk 2 above)
Comment (KevLehman) [reply apps/meteor/server/settings/lib/auditedSettingUpdates.ts:32]: Wondering if its a (hard) requirement cause it would be much easier to just throw some * in there without having to match the exact length

File: apps/meteor/server/settings/lib/auditedSettingUpdates.spec.ts [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+				{
+					id: settingId,
+					previous: '123**',
+					current: '678**',
+				},
+				{
+					type: 'user',
+					...actor,
+				},
+			);
```
[the remark this answers is not in the document]
Comment (julio-rocketchat) [reply apps/meteor/server/settings/lib/auditedSettingUpdates.spec.ts:304]: Shouldn't be an issue. Tests are passing and the fix is working as intended.

**Review by KevLehman — APPROVED** [review]

**Review by julio-rocketchat — DISMISSED** [review]

**Comment by dougfabris** [comment]
QQ, instead of a brand new label `do not merge`, wouldn't be better to convert it to draft? 
Isn't safer to prevent the merge?
